src/SprykerEco/Zed/Unzer/Persistence/UnzerRepository.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Persistence;
+
+use Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionCollectionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionConditionsTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionCriteriaTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransfer;
+use Generated\Shared\Transfer\StoreRelationTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCollectionTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsConditionsTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCriteriaTransfer;
+use Generated\Shared\Transfer\UnzerCustomerTransfer;
+use Orm\Zed\Unzer\Persistence\SpyPaymentUnzerTransactionQuery;
+use Orm\Zed\Unzer\Persistence\SpyUnzerCredentialsQuery;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+
+/**
+ * @method \SprykerEco\Zed\Unzer\Persistence\UnzerPersistenceFactory getFactory()
+ */
+class UnzerRepository extends AbstractRepository implements UnzerRepositoryInterface
+{
+    /**
+     * @param string $orderReference
+     *
+     * @return \Generated\Shared\Transfer\PaymentUnzerTransfer|null
+     */
+    public function findPaymentUnzerByOrderReference(string $orderReference): ?PaymentUnzerTransfer
```

Comment:
we have 3 similar methods `findPaymentUnzerBy...`
could be reworked to a single method to use Criteria/Conditions